src/MDMTransaction.h

```diff
@@ -27,9 +27,36 @@ NS_SWIFT_NAME(Transaction)
 
 #pragma mark Adding plans to a transaction
 
-/** Associate an plan with a given target. */
+/**
+Associates a plan with a given target.
+
+@param plan The plan to add to this transaction.
+@param target The target on which the plan can operate.
+*/
+- (void)addPlan:(nonnull id<MDMPlan>)plan
+       toTarget:(nonnull id)target
+  NS_SWIFT_NAME(add(plan:to:));
+
+/**
+ Associates a named plan with a given target.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan.
+ */
 - (void)addPlan:(nonnull id<MDMPlan>)plan
        toTarget:(nonnull id)target
-    NS_SWIFT_NAME(add(plan:to:));
+       withName:(nonnull NSString *)name
+  NS_SWIFT_NAME(add(plan:to:withName:));
+
+/**
+ Removes a named plan from the transaction.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan
+ */
+- (void)removePlanNamed:(nonnull NSString *)name
+  NS_SWIFT_NAME(remove(name:));
```

Comment:
In https://material-motion.gitbooks.io/material-motion-starmap/content/specifications/runtime/named-plans.html the "Remove-then-add" section explains the expected behavior of addPlan/removePlan for named plans on a given target.

Notably, for a given target there should only be one performer acting on on plan for a given name. The type of performer might change when a new plan is added, so when a named plan is associated with a target we need to give the previously-registered performer an opportunity to release any state associated with the name.